src/data/polynomial/div.lean

```diff
@@ -395,11 +395,14 @@ lemma mul_div_by_monic_eq_iff_is_root : (X - C a) * (p /ₘ (X - C a)) = p ↔ i
   by conv {to_rhs, rw ← mod_by_monic_add_div p (monic_X_sub_C a)};
     rw [mod_by_monic_X_sub_C_eq_C_eval, h, C_0, zero_add]⟩
 
-lemma dvd_iff_is_root : (X - C a) ∣ p ↔ is_root p a :=
+lemma dvd_iff_is_root : X - C a ∣ p ↔ is_root p a :=
 ⟨λ h, by rwa [← dvd_iff_mod_by_monic_eq_zero (monic_X_sub_C _),
     mod_by_monic_X_sub_C_eq_C_eval, ← C_0, C_inj] at h,
   λ h, ⟨(p /ₘ (X - C a)), by rw mul_div_by_monic_eq_iff_is_root.2 h⟩⟩
 
+lemma X_sub_C_dvd_sub_C_eval : X - C a ∣ p - C (p.eval a) :=
+by rw [dvd_iff_is_root, is_root, eval_sub, eval_C, sub_self]
```

Comment:
I guess this also follows from something like `q | p.comp q` with `q = X - C a`? Is that statement true, and do we have it?

Reply:
I think the correct statement is `(q - r) | (p.comp q - p.comp r)` and it follows from [polynomial.comp_eq_sum_left](https://leanprover-community.github.io/mathlib_docs/data/polynomial/eval.html#polynomial.comp_eq_sum_left) and [sub_dvd_pow_sub_pow](https://leanprover-community.github.io/mathlib_docs/algebra/geom_sum.html#sub_dvd_pow_sub_pow), but `(p.comp q - p.comp r)` is not `p.comp (q-r)`. I can't find the exact statement but [polynomial.sub_dvd_eval_sub](https://leanprover-community.github.io/mathlib_docs/data/polynomial/div.html#polynomial.sub_dvd_eval_sub) is similar.